__tests__/example.test.ts

```diff
@@ -3,25 +3,25 @@ import * as testlib from '../src/validate';
 import * as testreq from '../src/requests';
 import * as fakeapp from './fake-app';
 
+const requestSync = testreq.generateSyncRequest();
+const responseSync = fakeapp.onSync(requestSync);
```

Comment:
This thought came across to me as well, but defined as a global constant since the Query request takes in the Sync data as well to identify the traits needed. Maybe recalling the response sync is an option in the query test instead?

Reply:
I think it probably showcases the use case better if you do it all in the context of the test. If SYNC is required to evaluate QUERY, then have the QUERY test gather the SYNC response first before feeding it into the QUERY validation. Otherwise, it looks like your holding onto magic data just because your validator needs it.